txn_test.go

```diff
@@ -2178,3 +2178,53 @@ func TestTxn_Changes(t *testing.T) {
 		})
 	}
 }
+
+func TestTxn_GetIterAndDelete(t *testing.T) {
+	schema := &DBSchema{
+		Tables: map[string]*TableSchema{
+			"main": {
+				Name: "main",
+				Indexes: map[string]*IndexSchema{
+					"id": {
+						Name:    "id",
+						Unique:  true,
+						Indexer: &StringFieldIndex{Field: "ID"},
+					},
+					"foo": {
+						Name:    "foo",
+						Indexer: &StringFieldIndex{Field: "Foo"},
+					},
+				},
+			},
+		},
+	}
+	db, err := NewMemDB(schema)
+	assertNilError(t, err)
+
+	key := "aaaa"
+	txn := db.Txn(true)
+	assertNilError(t, txn.Insert("main", &TestObject{ID: "1", Foo: key}))
+	assertNilError(t, txn.Insert("main", &TestObject{ID: "123", Foo: key}))
+	assertNilError(t, txn.Insert("main", &TestObject{ID: "2", Foo: key}))
+	txn.Commit()
+
+	txn = db.Txn(true)
+	// Delete something
+	assertNilError(t, txn.Delete("main", &TestObject{ID: "123", Foo: key}))
+
+	iter, err := txn.Get("main", "foo", key)
+	assertNilError(t, err)
+
+	for obj := iter.Next(); obj != nil; obj = iter.Next() {
```

Comment:
Without this code change, this test would panic here on `iter.Next()` the second iteration.